### PR TITLE
Tensor Product Basis

### PR DESCRIPTION
**Pull request by quialst**
## Description

Added a tensor product class which takes an existing basis object and creates a basis of the tensor product using the product of the basis elements.

## Linked Issue
Fixes # (issue number)

## Implementation Checklist
- [ ] My class inherits from `PolynomialBasis` (if applicable).
- [ ] I have implemented all abstract methods (`evaluate_basis`, `fit`, etc.).
- [ ] I have used `numpy` vectorization to avoid inefficient Python loops.
- [ ] Documentation strings (docstrings) are provided for all public methods.

### 1. Mathematical Integrity
- [ ] **Convergence:** Does the implementation maintain expected accuracy? (Checked for Runge's phenomenon or stability).
- [ ] **Edge Cases:** Does it handle single points or non-equally spaced nodes correctly?
- [ ] **Domain Mapping:** Does the class correctly map the input domain to the basis domain (e.g., [-1, 1] for Chebyshev)?

### 2. API & Style
- [ ] **Method Signatures:** Methods match the Abstract Base Class exactly.
- [ ] **Naming:** Follows `CamelCase` for classes and `snake_case` for functions/variables.
- [ ] **Clean Code:** No "magic numbers"; constants are named and explained.

### 3. Testing Quality
- [ ] **Coverage:** Unit tests cover typical usage and edge cases.
- [ ] **Precision:** Tests use `np.testing.assert_allclose` instead of `==`.
- [ ] **Validation:** There is a test proving the basis can exactly reconstruct a polynomial of its own degree.

## Screenshots / Plots

### CODE REVIEW

File: src/numanalysislib/basis/tensor.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        self.bx = bx
+        self.by = by
+        self.nx = bx.n_dofs
+        self.ny = by.n_dofs
+        total_dofs = self.nx * self.ny
+
+        super().__init__(total_dofs - 1, bx.a, by.a)
+
+    def _unflatten_index(self, index: int):
+        """Helper function to loop through vectors"""
```
Comment (mtezzele) [inline src/numanalysislib/basis/tensor.py]: please improve the doc saying what you are doing. Also please use the proper style to describe parameters and return

**Comment by max-collins** [comment]
The tests look pretty coherent, but I'd be curious to see if the plotter functionality is available for bx and by separately.

File: src/numanalysislib/basis/tensor.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import numpy as np
+from numanalysislib.basis._abstract import PolynomialBasis
+
+class TensorProductBasis(PolynomialBasis):
+    def __init__(self, bx: PolynomialBasis, by: PolynomialBasis):
```
Comment (mtezzele) [inline src/numanalysislib/basis/tensor.py:5]: you need to document the arguments. It is not so obvious that bx and by stand for basis_x, basis_y (I suppose)

**Comment by mtezzele** [comment]
Please add a plotting functionality
